src/crypto/hash.h

```diff
@@ -22,6 +22,20 @@ namespace crypto
 
   /** Create a default incremental SHA256 hash provider */
   std::shared_ptr<ISha256Hash> make_incremental_sha256();
+
+  inline std::vector<uint8_t> hkdf(
+    MDType md_type,
+    size_t length,
+    const std::vector<uint8_t>& ikm,
+    const std::vector<uint8_t>& salt = {},
+    const std::vector<uint8_t>& info = {})
+  {
+#if defined(CRYPTO_PROVIDER_IS_MBEDTLS) && defined(INSIDE_ENCLAVE)
```

Comment:
As discussed, I believe `&& defined(INSIDE_ENCLAVE)` should go before merging this PR?

Reply:
It would work on the CI machines, but necessarily on all dev VMs. I'll remove it and install a new one on my VM and if we get complaints that it doesn't build just send them my way :)

Reply:
We tell contributors to use the playbooks in this repo to setup their development environment (see https://microsoft.github.io/CCF/main/build_apps/build_setup.html), which installs a recent-enough version of mbedtls so I think we're good :)